### PR TITLE
Add FNet Preprocessor

### PR DESCRIPTION
**Pull request by abheesht17**
https://colab.research.google.com/drive/1wd7ApfnTwS_xqa62Isdx3CGIZ95ixwxC?usp=sharing

### CODE REVIEW

File: keras_nlp/models/fnet/fnet_tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    # Detokenization.
+    tokenizer.detokenize(tf.constant([[[2, 14, 2231, 886, 2385, 3]]))
+    ```
+    """
+
+    def __init__(self, proto, **kwargs):
+        super().__init__(proto=proto, **kwargs)
+
+        # Check for necessary special tokens.
+        cls_token = "[CLS]"
```
Comment (mattdangerw) [inline keras_nlp/models/fnet/fnet_tokenizer.py:70]: This is odd, they really mix special tokens styles like this?

I guess this is half BERT style, half sentencepiece defaults.

Comment (abheesht17) [reply keras_nlp/models/fnet/fnet_tokenizer.py:70]: Yeah, I verified it with HF's tokenizer output

File: keras_nlp/models/fnet/fnet_preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+class FNetPreprocessor(Preprocessor):
+    """An FNet preprocessing layer which tokenizes and packs inputs.
+
+    This preprocessing layer will do three things:
+
+     - Tokenize any number of input segments using the `tokenizer`.
+     - Pack the inputs together using a `keras_nlp.layers.MultiSegmentPacker`.
+       with the appropriate `"[CLS]"`, `"[SEP]"` and `"<pad>"` tokens.
+     - Construct a dictionary with keys `"token_ids"`, `"segment_ids"` and
+       `"padding_mask"`, that can be passed directly to
```
Comment (mattdangerw) [inline keras_nlp/models/fnet/fnet_preprocessor.py:38]: there is no "padding_mask" right?

**Review by mattdangerw — APPROVED** [review]
Just one nit!

**Pull request merged**